src/fastoad/model_base/flight_point.py

```diff
@@ -59,78 +73,127 @@ class FlightPoint:
 
         Python allows to add attributes to any instance at runtime, but for FlightPoint to run
         smoothly, especially when exchanging data with pandas, you have to work at class level.
-        This can be done using :meth:`add_field`, preferably outside of any class or function::
+        This can be done using :meth:`add_field`, preferably outside any class or function::
 
-            # Adds a float field with None as default value
-            >>> FlightPoint.add_field("ion_drive_power")
+            # Adding a float field with None as default value
+            >>> FlightPoint.add_field(
+            ...    "ion_drive_power",
+            ...    unit="W",
+            ...    is_cumulative=False, # Tells if quantity sums up during mission
+            ...    is_output=True, # Tells if quantity is expected as mission output
+            ...    )
 
-            # Adds a field and define its type and default value
+            # Adding a field and defining its type and default value
             >>> FlightPoint.add_field("warp", annotation_type=int, default_value=9)
 
             # Now these fields can be used at instantiation
             >>> fp = FlightPoint(ion_drive_power=110.0, warp=12)
 
-            # Removes a field, even an original one (useful only to avoid having it in outputs)
+            # Removing a field, even an original one
             >>> FlightPoint.remove_field("sfc")
 
     .. note::
 
-        All parameters in FlightPoint instances are expected to be in SI units.
+        All original parameters in FlightPoint instances are expected to be in SI units.
 
     """
 
-    time: float = 0.0  #: Time in seconds.
-    altitude: float = None  #: Altitude in meters.
-    isa_offset: float = 0.0  #: temperature deviation from Standard Atmosphere
-    ground_distance: float = 0.0  #: Covered ground distance in meters.
-    mass: float = None  #: Mass in kg.
-    consumed_fuel: float = 0.0  #: Consumed fuel since mission start, in kg.
-    true_airspeed: float = None  #: True airspeed (TAS) in m/s.
-    equivalent_airspeed: float = None  #: Equivalent airspeed (EAS) in m/s.
-    mach: float = None  #: Mach number.
-    engine_setting: EngineSetting = None  #: Engine setting.
+    #: Time in seconds.
+    time: float = field(
+        default=0.0, metadata={FIELD_DESCRIPTOR: _FieldDescriptor(is_cumulative=True, unit="s")}
+    )
```

Comment:
Is this added complexity necessary or is it just cleaner to do it this way ?

Reply:
Not necessary. I could have use `is_cumulative`, `is_output` and `unit` directly as fields of the metadata dict.
But the current way offers a less error-prone way to populate the metadata, and also allows having default values for theses 3 fields.